clang/lib/Headers/hlsl/hlsl_intrinsics.h

```diff
@@ -11,557 +11,614 @@
 
 namespace hlsl {
 
-__attribute__((availability(shadermodel, introduced = 6.0)))
-__attribute__((clang_builtin_alias(__builtin_hlsl_wave_active_count_bits))) uint
-WaveActiveCountBits(bool bBit);
+// Note: Functions in this file are sorted alphabetically, then grouped by base
+// element type, and the element types are sorted by size, then singed integer,
+// unsigned integer and floating point. Keeping this ordering consistent will
+// help keep this file manageable as it grows.
 
+#define _HLSL_BUILTIN_ALIAS(builtin)                                           \
```

Comment:
Is it ok to have macros that start with an underscore? I thought identifiers starting with _ are reserved in c/c++, but not sure about macros.

Also, since this is a header file should we undef them at the end? Not sure we expect others to use the macros themselves, but more of just an implementation detail of the header.

Reply:
Yes, this is definitely a reserved identifier. Anything with two underscores in the name and anything starting with an underscore and upper letter is reserved. There are also some more complicated rules, but that's not relevant here.

Reply:
It being reserved is the point here. `_<capital letter>` macros are reserved for use by the compiler. This header is part of the compiler implementation. It ships with the compiler and is tied to it.

We could alternatively move this definition into the actual compiler code (like under InitPreprocessor), but that would be harder to maintain. We do similar things in the `__cuda_*` headers in the parent directory of this header as well as the target-specific intrinsic headers (see: https://github.com/llvm/llvm-project/blob/main/clang/lib/Headers/xmmintrin.h#L2076)